### PR TITLE
[velbus] New module VMBELPIR + Improvement clock alarms management

### PR DESCRIPTION
**Pull request by Rosen01**
New functionnalities :

1.  Add the module VMBELPIR (#12279)

2. Improves the clocks alarm management (#11655)
- Global alarms are now set from the bridge thing.
- Local alarms are still set from the module thing, but the alarm type channels on modules are now readonly.
- Reduces the flooding of the bus by adding a delay of 10 seconds before sending the update on the bus :
_Removes the send on the bus after each update of a clock alarm value.
Sends only the values of the last update after 10 seconds._

Sources : https://github.com/Rosen01/openhab-addons/releases

Signed-off-by: Daniel Rosengarten <github@praetorians.be>

### CODE REVIEW

**Review by cedricboon — APPROVED** [review]
Thanks for these great improvements, lgtm!

File: bundles/org.openhab.binding.velbus/src/main/java/org/openhab/binding/velbus/internal/handler/VelbusBridgeHandler.java [hunk 1]
```diff
@@ -68,6 +75,32 @@
 
     private boolean listenerStopped;
 
+    private final ChannelUID clockAlarm1Enabled = new ChannelUID(thing.getUID(), "bridgeClockAlarm",
+            "clockAlarm1Enabled");
```
Comment (lolodomo) [inline bundles/org.openhab.binding.velbus/src/main/java/org/openhab/binding/velbus/internal/handler/VelbusBridgeHandler.java:79]: Rather work with channel IDs and define these IDs as constants in VelbusBindingConstants. This is the usual way to do.

File: bundles/org.openhab.binding.velbus/src/main/resources/OH-INF/thing/thing-types.xml [hunk 2]
```diff
@@ -531,6 +537,25 @@
 		<config-description-ref uri="thing-type:velbus:33channelDeviceWithTemperatureSensor"/>
 	</thing-type>
 
+	<thing-type id="vmbelpir">
```
Comment (lolodomo) [inline bundles/org.openhab.binding.velbus/src/main/resources/OH-INF/thing/thing-types.xml:540]: README should be updated.
Default properties should be updated (velbus.properties) using the maven command.

**Review by lolodomo — CHANGES_REQUESTED** [review]
Review part 1 of 2

File: bundles/org.openhab.binding.velbus/src/main/resources/OH-INF/thing/thing-types.xml (hunk 2 above)
Comment (Rosen01) [reply bundles/org.openhab.binding.velbus/src/main/resources/OH-INF/thing/thing-types.xml:540]: Can you give me the maven command to do the update of the default properties ?

File: bundles/org.openhab.binding.velbus/src/main/java/org/openhab/binding/velbus/internal/handler/VelbusBridgeHandler.java (hunk 1 above)
Comment (Rosen01) [reply bundles/org.openhab.binding.velbus/src/main/java/org/openhab/binding/velbus/internal/handler/VelbusBridgeHandler.java]: Changes done.

File: bundles/org.openhab.binding.velbus/src/main/resources/OH-INF/thing/thing-types.xml (hunk 2 above)
Comment (Rosen01) [reply bundles/org.openhab.binding.velbus/src/main/resources/OH-INF/thing/thing-types.xml:540]: Changes done.

File: bundles/org.openhab.binding.velbus/src/main/java/org/openhab/binding/velbus/internal/handler/VelbusBridgeHandler.java [hunk 3]
```diff
@@ -141,7 +155,87 @@ public void dispose() {
...
+                        alarmClock.setBedtimeMinute(bedTimeMinute);
+                    }
+                    break;
+                }
+            }
+
+            if (alarmNumber == 1)
+                lastUpdateAlarm1TimeMillis = System.currentTimeMillis();
+            else
+                lastUpdateAlarm2TimeMillis = System.currentTimeMillis();
```
Comment (lolodomo) [inline bundles/org.openhab.binding.velbus/src/main/java/org/openhab/binding/velbus/internal/handler/VelbusBridgeHandler.java:210]: Please surround each branch with `{}`

File: bundles/org.openhab.binding.velbus/src/main/java/org/openhab/binding/velbus/internal/handler/VelbusBridgeHandler.java [hunk 4]
```diff
@@ -141,7 +155,87 @@ public void dispose() {
...
+        }
+    }
+
+    public synchronized void sendAlarmPacket(int alarmNumber, byte[] packetBytes) {
+        long timeSinceLastUpdate;
+
+        if (alarmNumber == 1)
+            timeSinceLastUpdate = System.currentTimeMillis() - lastUpdateAlarm1TimeMillis;
+        else
+            timeSinceLastUpdate = System.currentTimeMillis() - lastUpdateAlarm2TimeMillis;
```
Comment (lolodomo) [inline bundles/org.openhab.binding.velbus/src/main/java/org/openhab/binding/velbus/internal/handler/VelbusBridgeHandler.java:232]: Please surround each branch with `{}`

File: bundles/org.openhab.binding.velbus/src/main/java/org/openhab/binding/velbus/internal/handler/VelbusBridgeHandler.java [hunk 5]
```diff
@@ -141,7 +155,87 @@ public void dispose() {
...
+
+        // If a value of the alarm has been updated, discard this old update
+        if (timeSinceLastUpdate < 10000)
+            return;
```
Comment (lolodomo) [inline bundles/org.openhab.binding.velbus/src/main/java/org/openhab/binding/velbus/internal/handler/VelbusBridgeHandler.java:236]: Please surround each branch with `{}`

File: bundles/org.openhab.binding.velbus/src/main/java/org/openhab/binding/velbus/internal/handler/VelbusBridgeHandler.java [hunk 6]
```diff
@@ -141,7 +155,87 @@ public void dispose() {
...
+
+            VelbusSetLocalClockAlarmPacket packet = new VelbusSetLocalClockAlarmPacket((byte) 0x00, alarmNumber,
+                    alarmClock);
+            byte[] packetBytes = packet.getBytes();
+
+            // Schedule the send of the packet to see if there is another update in less than 10 secondes (reduce
+            // flooding of the bus)
+            scheduler.schedule(() -> {
+                sendAlarmPacket(alarmNumber, packetBytes);
+            }, 10000, TimeUnit.MILLISECONDS);
```
Comment (lolodomo) [inline bundles/org.openhab.binding.velbus/src/main/java/org/openhab/binding/velbus/internal/handler/VelbusBridgeHandler.java:220]: 10000 could be set as a constant.

**Review by lolodomo — CHANGES_REQUESTED** [review]
Just small final comments

File: bundles/org.openhab.binding.velbus/src/main/java/org/openhab/binding/velbus/internal/handler/VelbusBridgeHandler.java (hunk 3 above)
Comment (Rosen01) [reply bundles/org.openhab.binding.velbus/src/main/java/org/openhab/binding/velbus/internal/handler/VelbusBridgeHandler.java]: Done.

File: bundles/org.openhab.binding.velbus/src/main/java/org/openhab/binding/velbus/internal/handler/VelbusBridgeHandler.java (hunk 4 above)
Comment (Rosen01) [reply bundles/org.openhab.binding.velbus/src/main/java/org/openhab/binding/velbus/internal/handler/VelbusBridgeHandler.java]: Done.

File: bundles/org.openhab.binding.velbus/src/main/java/org/openhab/binding/velbus/internal/handler/VelbusBridgeHandler.java (hunk 5 above)
Comment (Rosen01) [reply bundles/org.openhab.binding.velbus/src/main/java/org/openhab/binding/velbus/internal/handler/VelbusBridgeHandler.java]: Done.

File: bundles/org.openhab.binding.velbus/src/main/java/org/openhab/binding/velbus/internal/handler/VelbusBridgeHandler.java (hunk 6 above)
Comment (Rosen01) [reply bundles/org.openhab.binding.velbus/src/main/java/org/openhab/binding/velbus/internal/handler/VelbusBridgeHandler.java]: Done.

**Review by lolodomo — APPROVED** [review]
LGTM

**Comment by lolodomo** [comment]
@cedricboon : would you like to review these changes?

**Comment by cedricboon** [comment]
LGTM

**Pull request merged**